src/Npgsql/Replication/Logical/Protocol/NpgsqlLogicalReplicationConnectionExtensions.cs

```diff
@@ -0,0 +1,352 @@
+        /// first command run in that transaction. Finally, <see cref="SlotSnapshotInitMode.NoExport"/> will just use
+        /// the snapshot for logical decoding as normal but won't do anything else with it.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// The token to monitor for cancellation requests.
+        /// The default value is <see cref="CancellationToken.None"/>.
+        /// </param>
+        /// <returns>
+        /// A <see cref="NpgsqlPgOutputReplicationSlot"/> that wraps the newly-created replication slot.
+        /// </returns>
+        [PublicAPI]
+        public static async Task<NpgsqlPgOutputReplicationSlot> CreateReplicationSlot(
+            this NpgsqlLogicalReplicationConnection connection,
+            string slotName,
+            bool temporarySlot = false,
+            SlotSnapshotInitMode? slotSnapshotInitMode = null,
+            CancellationToken cancellationToken = default)
+        {
+            // We don't enter NoSynchronizationContextScope here since we (have to) do it in CreateReplicationSlotForPlugin, because
+            // otherwise it couldn't be set for external plugins.
+            var options =
+                await connection.CreateReplicationSlotForPlugin(slotName, "pgoutput", temporarySlot, slotSnapshotInitMode, cancellationToken);
+            return new NpgsqlPgOutputReplicationSlot(options);
+        }
+
+        /// <summary>
+        /// Instructs the server to start the Logical Streaming Replication Protocol (pgoutput logical decoding plugin),
+        /// starting at WAL location <paramref name="walLocation"/> or at the slot's consistent point if <paramref name="walLocation"/>
+        /// isn't specified.
+        /// The server can reply with an error, for example if the requested section of the WAL has already been recycled.
+        /// </summary>
+        /// <param name="connection">The <see cref="NpgsqlLogicalReplicationConnection"/> to use for starting replication</param>
+        /// <param name="slot">The replication slot that will be updated as replication progresses so that the server
+        /// knows which WAL segments are still needed by the standby.
+        /// </param>
+        /// <param name="walLocation">The WAL location to begin streaming at.</param>
+        /// <param name="options">The collection of options passed to the slot's logical decoding plugin.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.
+        /// The default value is <see cref="CancellationToken.None"/>.</param>
+        /// <returns>A <see cref="Task{T}"/> representing an <see cref="IAsyncEnumerable{T}"/> that
+        /// can be used to stream WAL entries in form of <see cref="LogicalReplicationProtocolMessage"/> instances.</returns>
+        [PublicAPI]
+        public static Task<IAsyncEnumerable<LogicalReplicationProtocolMessage>> StartReplication(
```

Comment:
It seems like the two methods just below can be merged into this one (nobody else calls them). If we do this, the public StartReplication can be an async-iterator method directly, simply just returning IAsyncEnumerable<T> directly, and use [EnumeratorCancellation]. It would be following the "improved" method described in http://blog.monstuff.com/archives/2019/03/async-enumerables-with-cancellation.html.